src/Providers/ValidationProvider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+
+namespace Wepesi\App\Providers;
+
+
+use Wepesi\App\Providers\Contracts\Contracts;
+
+abstract class ValidationProvider implements Contracts
+{
+    abstract function min($rule);
+    abstract function max($rule);
+    abstract function required();
+    /**
+     *
+     * @param array $value
+     * @return void
+     */
+    function addError(array $value): void
```

Comment:
add to `addError` and `result` method to be re-used by all class that inherit the class and avoid duplication